blocksync/pool.go

```diff
@@ -274,6 +320,10 @@ func (pool *BlockPool) RemovePeerAndRedoAllPeerRequests(height int64) p2p.ID {
 	defer pool.mtx.Unlock()
 
 	request := pool.requesters[height]
+	if request == nil {
+		// this shouldn't happen, but to be on the safe side, let's ignore
```

Comment:
make sense can we log this as an error if this isn't supposed to happen so that we can be notified if it does occur?